### PR TITLE
Add associated arbitrary metadata to Projects

### PR DESCRIPTION
**Pull request by Bialogs**

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -87,6 +87,14 @@
     t.index ["user_id"], name: "index_project_members_on_user_id"
   end
 
+  create_table "project_metadata", force: :cascade do |t|
+    t.json "data", null: false
```
Comment (jkufro) [inline db/schema.rb:91]: What is the reason for not making a json data field on the project table itself?

Comment (Bialogs) [reply db/schema.rb:91]: No need to make the default Project select large if it has a lot of metadata

File: app/models/project.rb [hunk 2]
```diff
@@ -7,6 +7,7 @@ class Project < ApplicationRecord
   has_many :project_members, dependent: :destroy
   has_many :users, through: :project_members
   has_many :rules, dependent: :destroy
+  has_one :project_metadata, dependent: :destroy
```
Comment (jkufro) [inline app/models/project.rb:11]: When controllers are implemented for managing projects we will probably want `accepts_nested_attributes_for :project_metadata` on the Project model

https://api.rubyonrails.org/classes/ActiveRecord/NestedAttributes/ClassMethods.html

Comment (jkufro) [reply app/models/project.rb:11]:
```suggestion
accepts_nested_attributes_for :project_metadata
```

**Review by jkufro — CHANGES_REQUESTED** [review]
see inline comment about nested attributes

File: app/models/project.rb [hunk 3]
```diff
@@ -7,6 +7,8 @@ class Project < ApplicationRecord
...
+accepts_nested_attributes_for :project_metadata
```
Comment (jkufro) [inline app/models/project.rb:11]: indentation issue was my fault when creating the suggestion

**Review by jkufro — APPROVED** [review]

**Pull request merged**